### PR TITLE
Fix defect that button is immediately enabled after initiating long jobs

### PR DESCRIPTION
**Pull request by ke-yu**
This pull request is identical to PR https://github.com/DynamoDS/Dynamo/pull/3594, somehow the original one can't be merged automatically by github.

The problem happens when loading a dyn file and run it. 

Run button's ``IsEnabled`` binds to view model's ``RunEnabled`` which gets its value from home workspace model's ``RunEnabled``. View model registers property change event for model's current workspace model in its constructor, so that if home workspace's ``RunEnabled`` is updated, Run button will be enabled/disabled accordingly. But that means view model never handles property change event if current workspace model is changed, for example, loading a new .dyn file. 

As the owner of workspace model is model, it is not good idea that view model directly handle events from workspace model. Instead, view model monitors model's property changed event. Therefore if ``RunEnable`` property is changed, the event handling chain is home workspace model -> model -> view model -> Run button. 

@Benglin  please review the change...

### CODE REVIEW

File: src/DynamoCore/Models/HomeWorkspaceModel.cs [hunk 1]
```diff
@@ -159,6 +146,7 @@ public override void Clear()
         {
             base.Clear();
             PreloadedTraceData = null;
+            RunEnabled = true;
```
Comment (Benglin) [inline src/DynamoCore/Models/HomeWorkspaceModel.cs]: Do we not want to conditionally exclude this with `#if DEBUG`?

Comment (ke-yu) [inline src/DynamoCore/Models/HomeWorkspaceModel.cs]: Why? RunEnabled for Run button, not Run automatically option box.

Comment (Benglin) [inline src/DynamoCore/Models/HomeWorkspaceModel.cs]: Oops, I got that mixed up. Decision fatigue towards the end of the day.

**Comment by Benglin** [comment]
LGTM!

**Pull request merged**